lib/order.rb

```diff
@@ -0,0 +1,51 @@
+class Order
+  
+  ERR_INVALID_ORDER = "The selected dish is not on the menu".freeze
+  ERR_EMPTY_ORDER = "The order is empty".freeze
+  
+  attr_reader :dishes, :total
+  
+  def initialize(menu)
+    @dishes = {}
+    @total = 0
+    @menu = menu
+  end
+  
+  def add(dish, qty = 1)
+    fail ERR_INVALID_ORDER unless menu.contains? dish
+    dishes[dish] = qty
+  end
+  
+  alias update add
+  
+  def remove(dish)
+    dishes.delete dish if contains?(dish)
+  end
+  
+  def total
```

Comment:
you already have an attire_reader :total, that is being overwritten by this method